composer/lib/dependabot/composer/helpers.rb

```diff
@@ -174,7 +174,7 @@ def self.invalid_v2_requirement?(composer_json)
 
       sig { params(dependency_url: String).returns(String) }
       def self.clean_dependency_url(dependency_url)
-        return dependency_url unless URI::DEFAULT_PARSER.regexp[:ABS_URI].match?(dependency_url)
```

Comment:
Not 100% sure why `URI::RFC_3986.make_regexp.match?(dependency_url)` doesn't give the same results for this specific instance. But using `RFC2396_PARSER` explicitly works for now.